.github/workflows/pr-realm-js.yml

```diff
@@ -0,0 +1,391 @@
+          path: |
+            dependencies.list
+            ${{ matrix.variant.artifact-path }}
+
+  publish:
+    runs-on: ubuntu-20.04
+    needs: [build]
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Setup node version
+        uses: actions/setup-node@v2
+        with:
+          node-version: '14'
+          cache: 'npm'
+          cache-dependency-path: '**/package-lock.json'
+
+      - name: Setup NPM version
+        run: npm install -g npm@7
+
+      - name: Get NPM cache directory
+        id: npm-cache-dir
+        run: echo "::set-output name=dir::$(npm config get cache)"
+
+      - name: Restore NPM cache
+        id: npm-cache
+        uses: actions/cache@v2
+        with:
+          path: ${{ steps.npm-cache-dir.outputs.dir }}
+          key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-node-
+
+      - name: Install node modules
+        run: npm ci --ignore-scripts
+
+      - name: Download prebuilds
+        uses: actions/download-artifact@v2
+        with:
+          name: realm-js-prebuilds
+
+      - name: Display structure of downloaded files
+        run: ls -R prebuilds react-native/android/src/main/jniLibs
+
+      - name: Publish to Github
+        run: npx prebuild --upload-all ${{ secrets.GITHUB_TOKEN }}
+
+  integration-tests:
+    name: Integration tests (${{matrix.variant.target}}) for ${{ matrix.variant.environment }} on ${{ matrix.variant.os }}
+    needs: [build]
+    env:
+      REALM_DISABLE_ANALYTICS: 1
+      MOCHA_REMOTE_TIMEOUT: 60000
+      REALM_BASE_URL: ${{ secrets.REALM_QA_BASE_URL }}
+      REALM_PUBLIC_KEY: ${{ secrets.ATLAS_QA_PUBLIC_API_KEY }}
+      REALM_PRIVATE_KEY: ${{ secrets.ATLAS_QA_PRIVATE_API_KEY }}
+      MOCHA_REMOTE_REPORTER: mocha-github-actions-reporter
+      MOCHA_REMOTE_EXIT_ON_ERROR: true
+      SPAWN_LOGCAT: true
+      HEADLESS_DEBUGGER: true
+      # RETRIES: 5
+      # RETRY_DELAY: 1800000 # 30 min
+      # Pin the Xcode version
+      DEVELOPER_DIR: /Applications/Xcode_12.5.1.app
+      ANDROID_NDK_HOME: /usr/local/lib/android/sdk/ndk/21.4.7075529
+    runs-on: ${{ matrix.variant.runner }}
+    strategy:
+      fail-fast: false
+      matrix:
+        variant:
+          #type: [Release, Debug]
+          - { os: linux, target: test, runner: ubuntu-20.04, environment: node }
+          - { os: linux, target: "test:main", runner: ubuntu-20.04, environment: electron }
+          - { os: linux, target: "test:renderer", runner: ubuntu-20.04, environment: electron }
+          #- { os: windows, target: test, runner: windows-2019, environment: node}
+          #- { os: windows, target: "test:main", runner: windows-2019, environment: electron }
+          #- { os: windows, target: "test:renderer", runner: windows-2019, environment: electron }
+          - { os: darwin, target: "test:main", runner: macos-11, environment: electron }
```

Comment:
Consider running sync tests only on certain environments (or just when merged to master).  Make a ticket